source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -39,6 +39,32 @@ struct ExtProcFilterStats {
   ALL_EXT_PROC_FILTER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
+inline constexpr absl::string_view ExtProcGrpcStatsName = "ext-proc-filter-state";
+
+class ExtProcGrpcStats : public Envoy::StreamInfo::FilterState::Object {
+public:
+  void record(std::chrono::microseconds latency, ProcessorState::CallbackState callback_state,
+              absl::Status call_status) {
+    switch (callback_state) {
+    case ProcessorState::CallbackState::Idle:
+      header_call_latencies.emplace_back(latency, call_status);
```

Comment:
The CallbackState::Idle state is when the filter is "Not waiting for anything", so ISTM a record() call in this state would be unexpected. Given that, I don't think we should be adding to header_call_latencies in this case. Instead, this could ENVOY_BUG or ENVOY_LOG(error).

Reply:
Thanks, I add an assertion in the caller.